config.go

```diff
@@ -101,6 +107,10 @@ type Config struct {
 	// LocalAddr is the local address to use when dialing an nsqd.
 	// If empty, a local address is automatically chosen.
 	LocalAddr net.Addr `opt:"local_addr"`
+	// Dialer affect connection when dialing an nsqd. Overwrite this to connect over proxy.
+	//
+	// Conflict with options LocalAddr and DialTimeout.
```

Comment:
Correct, this does conflict. We should probably error when a `dialer` and either of those two config fields are set.